File: Python/Jugador.py
from Dado import *


class Jugador():
    __nombre = ''
    __jugadasGanadas = 0
    __jugadasPerdidas = 0
    __jugadasEmpatadas = 0

    def __init__(self, nombre):
        self.__nombre = nombre

    def incrementarJugadasGanadas(self):
        self.__jugadasGanadas += 1

    def incrementarJugadasPerdidas(self):
        self.__jugadasPerdidas += 1

    def incrementarJugadasEmpatadas(self):
        self.__jugadasEmpatadas += 1

    def lanzarDado(dado):
        dado = Dado()
        return dado.generarNumero()

    def getNombre(self):
        return self.__nombre

    def getJugadasGanadas(self):
        return self.__jugadasGanadas

    def getJugadasPerdidas(self):
        return self.__jugadasPerdidas

    def getJugadasEmpatadas(self):
        return self.__jugadasEmpatadas

    def determinarGanador(self, jugador, nombre):
        jugador = Jugador(nombre)
        if self.__jugadasGanadas > jugador.getJugadasGanadas():
            return self.__nombre
        elif self.__jugadasGanadas < jugador.getJugadasGanadas():
            return jugador.getNombre()
        else:
            return None


def main():
    nombreJugador1 = input("Ingrese el nombre del jugador1: ").replace(
        ' ', '').capitalize()
    nombreJugador2 = input("Ingrese el nombre del jugador2: ").replace(
        ' ', '').capitalize()
    jugador1 = Jugador(nombreJugador1)
    jugador2 = Jugador(nombreJugador2)
    dado = Dado()
    valorJugador1 = 0
    valorJugador2 = 0
    for i in range(5):
        valorJugador1 = jugador1.lanzarDado()
        valorJugador2 = jugador2.lanzarDado()
        if valorJugador1 > valorJugador2:
            jugador1.incrementarJugadasGanadas()
            jugador2.incrementarJugadasPerdidas()
        elif valorJugador1 < valorJugador2:
            jugador1.incrementarJugadasPerdidas()
            jugador2.incrementarJugadasGanadas()
        else:
            jugador1.incrementarJugadasEmpatadas()
            jugador2.incrementarJugadasEmpatadas()
    print("Estadisticas de: ", jugador1.getNombre(), "\nJugadas ganadas: ", jugador1.getJugadasGanadas(
    ), "\nJugadas perdidas: ", jugador1.getJugadasPerdidas(), "\nJugadas empatadas: ", jugador1.getJugadasEmpatadas())
    print("Estadisticas de: ", jugador2.getNombre(), "\nJugadas ganadas: ", jugador2.getJugadasGanadas(
    ), "\nJugadas perdidas: ", jugador2.getJugadasPerdidas(), "\nJugadas empatadas: ", jugador2.getJugadasEmpatadas())
    print(jugador1.determinarGanador(jugador2, nombreJugador2), " Es el ganador")


if __name__ == '__main__':
    main()

File: Java/Jugador.java
import java.util.Scanner;

public class Jugador {
    private String nombre;
    private int jugadasGanadas;
    private int jugadasPerdidas;
    private int jugadasEmpatadas;

    public Jugador(String nombre) {
        this.nombre = nombre;
    }

    public void incrementarJugadasGanadas() {
        jugadasGanadas++;
    }

    public void incrementarJugadasPerdidas() {
        jugadasPerdidas++;

    }

    public void incrementarJugadasEmpatadas() {
        jugadasEmpatadas++;
    }

    public int lanzarDado(Dado dado) {
        return dado.generarNumero();
    }

    public String getNombre() {
        return nombre;
    }

    public int getJugadasGanadas() {
        return jugadasGanadas;
    }

    public int getJugadasPerdidas() {
        return jugadasPerdidas;
    }

    public int getJugadasEmpatadas() {
        return jugadasEmpatadas;
    }

    public String determinarGanador(Jugador jugador) {
        if (jugadasGanadas > jugador.getJugadasGanadas()) {
            return nombre;
        } else if (jugadasGanadas < jugador.getJugadasGanadas()) {
            return jugador.getNombre();
        } else {
            return null;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Ingresa el nombre del jugador 1: ");
        String nombreJugador1 = scanner.next();
        System.out.println("Ingresa el nombre del jugador 2: ");
        String nombreJugador2 = scanner.next();
        Jugador jugador1 = new Jugador(nombreJugador1);
        Jugador jugador2 = new Jugador(nombreJugador2);
        Dado dado = new Dado();

        int valorjugador1, valorjugador2;
        for (int i = 1; i <= 5; i++) {
            valorjugador1 = jugador1.lanzarDado(dado);
            valorjugador2 = jugador2.lanzarDado(dado);

            if (valorjugador1 > valorjugador2) {
                jugador1.incrementarJugadasGanadas();
                jugador2.incrementarJugadasPerdidas();
            } else if (valorjugador1 < valorjugador2) {
                jugador1.incrementarJugadasPerdidas();
                jugador2.incrementarJugadasGanadas();
            } else {
                jugador1.incrementarJugadasEmpatadas();
                jugador2.incrementarJugadasEmpatadas();
            }
        }
        System.out.println(
                "Estadisticas de: " + jugador1.getNombre() + "\nJugadas ganadas: " + jugador1.getJugadasGanadas()
                        + "\nJugadas perdidas: " + jugador1.getJugadasPerdidas() + "\nJugadas empatadas: "
                        + jugador1.getJugadasEmpatadas());
        System.out.println(
                "Estadisticas de: " + jugador2.getNombre() + "\nJugadas ganadas: " + jugador2.getJugadasGanadas()
                        + "\nJugadas perdidas: " + jugador2.getJugadasPerdidas() + "\nJugadas empatadas: "
                        + jugador2.getJugadasEmpatadas());
        System.out.println(jugador1.determinarGanador(jugador2) + " Es el ganador");
        scanner.close();
    }
}